debug/custom-source.html

```diff
@@ -31,59 +33,71 @@
     hash: true
 });
 
-const tileSize = 256;
-const colors = ['#74a9cf', '#3690c0', '#0570b0', '#045a8d'];
+const tileSize = 512;
+const colors = ['red', 'green', 'blue', 'yellow'];
 let currentColor = colors[0];
 
 const canvas = document.createElement('canvas');
 canvas.width = canvas.height = tileSize;
 const context = canvas.getContext('2d', {willReadFrequently: true});
 
+class CustomSource {
+    constructor() {
+        this.id = 'custom-source';
+        this.type = 'custom';
+        this.tileSize = tileSize;
+    }
+
+    async loadTile({z, x, y}) {
+        context.fillStyle = currentColor;
+        context.fillRect(0, 0, tileSize, tileSize);
+        return canvas;
```

Comment:
Returning the same canvas seems to duplicate the last drawn result on all tiles: 

```js
const colors = ['rgba(255,0,0,0.25)', 'rgba(0,255,0,0.25)', 'rgba(0,0,255,0.25)', 'rgba(255,255,0,0.25)'];

class CustomSource {
    async loadTile({z, x, y}) {
        context.clearRect(0, 0, tileSize, tileSize);
        context.fillStyle = currentColor;
        context.fillRect(0, 0, tileSize, tileSize);
        context.fillStyle = 'black';
        context.font = '18px serif';
        context.fillText(`${x}/${y}/${z}`, tileSize/2, tileSize/2);
        return canvas;
    }
}
```

<img width="878" alt="Screen Shot 2022-09-08 at 9 56 35 AM" src="https://user-images.githubusercontent.com/572717/189182558-c3085595-085a-43b5-81bb-d9684f40e61a.png">